### ISSUE TITLE
Feature request: Add the ability to speed up and slow down replay of a track without changing pitch

### ISSUE THREAD

**Issue by lexBuright** [cmus/cmus#514]
# Previous discussion

I seem to remember seeing this discussed on a mailing list and the response was along the lines of "cmus isn't a sound editor - go use audacity / ardour" (https://sourceforge.net/p/cmus/mailman/message/33264024/)

# Rationale / why this isn't editing

An argument for having this within a playback tool (and integrated with that tool) is listening to lectures / podcasts.

I find it quite useful to take a lecture, play it at double or 2.5 times speed, until it starts saying something that requires that a little thought, at which point I'll slow it done, and possible replaying interesting parts. Anecdotal afk conversations suggest that other people do this.

This feature is implemented in _vlc_, however _vlc_'s remote control functionality isn't as good as cmus's, its curses display isn't as good (and can mess up formatting when you interact with it via a remote control), and it is impossible to do relative seeks in _cmus_ but not via _vlc_'s remote control.

# Bloat

There is a bloat / "do one thing well" argument against this. There might be ways of avoiding pulling this into cmus's core by providing/reusing some interface, though there's a question of whether the complexity cost of the interface exceeds the cost.

I'm not sure what sort of interface this would be, it's more than just a filter, since it also messes with timings.

* Opinions?

I might want to implement this (vlc really sucks and I listen to a lot of podcasts), but would prefer to implement something that will get accepted?

**Comment by inspell** [comment]
Hi,

I would give this +1 since I requested the feature the first time ; )
It's perfect for guitar lessons too. And I also don't think this feature is
related to the "editing" category. It will be so awesome to have it in
cmus, I will just love it. Since I don't have the capabilities to implement
it I will not rant about it and if someone wants to implement it that will
be great!

Thanks,
Vladimir

**Comment by NSBrianWard** [comment]
Yeah, I just wanna pipe up here that a feature like this would be super useful. I'm currently transcribing an interview and I love the "c" "left" "right" transport controls just for navigating around the audio I'm trying to transcribe. To be able to knock the speed down a notch or two would be ridiculously helpful. Like, I'm not necessarily wanting to make a slower flat file, I want to be able to adjust it almost on the fly as needed depending on the speaker's rate of talking.

**Comment by tahmmee** [comment]
burp

**Comment by paulolimac** [comment]
it is a good feature to listen podcasts.
We could speed up and listen them more fast.
Like MPV player with `[` and `]`, or Youtube with its `playback speed` config `<` and `>` (aka `0.25, 0.5, 0.75, normal, 1.25, 1.5, 1.75, 2`)

**Comment by amenl** [comment]
The flying spaghetti monster brought me here.

**Comment by kr37** [comment]
Yes, would love this for transcribing!

**Comment by kidsil** [comment]
Anyone who wants a quick fix meanwhile, I just use a simple script to quickly convert audio files' tempo (X1.5): 
https://gist.github.com/kidsil/3ac6bd771311f0e7798a2b1bb5254fe9

Sure it's not during playback, but it works fine for me.

**Comment by idvorkin** [comment]
This is a killer feature for me. Are there other TUI players that support this that folks know of?
